### ISSUE TITLE
`amx_int8` and `amx_bf16` not supported on v1.2

### ISSUE THREAD

**Issue by koleini** [intel/processwatch#19]
The tool (v1.2) is unable to identify recognize the category for `amx_int8` and `amx_bf16` flags.

**Comment by grahamwoodward** [comment]
A dup of issue #18 ?

**Comment by grahamwoodward** [comment]
Ah so it is, my mistake. This could by zydis issue then? Not process watch ? Or did this previously work?

**Comment by adgubrud** [comment]
@koleini  @grahamwoodward 
In the past I've been able to get AMX instruction counts. I was just revisiting this yesterday and found this setup to work. 

```bash
git clone https://github.com/intel/processwatch.git
cd processwatch
git checkout a9e8f87138efdb4afdd37ea67d8e2ed47438344b
deps/bpftool 
git submodule init 
git submodule update --recursive 
git submodule init 
git submodule update --recursive 
cd ../zydis 
git submodule init 
git submodule update --recursive
cd .. 
./build_deps.sh 
cd ..
./build.sh
```

**Comment by koleini** [comment]
@adgubrud would you please check if `amx_int8` and `amx_bf16` flags work for you?

**Comment by adgubrud** [comment]
@koleini maybe I misunderstood your original ask - I thought you were looking for AMX flags in general. Like you captured above I can get information for AMX_TILE but not a separation of amx_int8 vs. amx_bf16. I would guess the best way to go about that would be to use the `-m` flag in processwatch and you'd have to filter for instructions using AMX, then bucket them yourself for different data types.

**Comment by matthew-olson-intel** [comment]
Yes -- right now, in pre-1.2, using Zydis, it doesn't separate int8 and bf16 AMX instructions. I think it should! I think we should consolidate our efforts and fix #18 too, by simply adding support for those categories to Capstone.

**Comment by matthew-olson-intel** [comment]
Update: I'm working on updating Capstone's x86 support. It looks like they've got some auto-updating scripts, but I also don't mind adding AMX manually if need be. There aren't that many AMX instructions.

**Comment by matthew-olson-intel** [comment]
Update: I'm working on adding AMX support to Capstone. They use LLVM TableGen to update their x86 instructions.

https://github.com/capstone-engine/capstone/issues/2503

**Comment by matthew-olson-intel** [comment]
Scratch that: Capstone folks seem to want to simply call Zydis to do their x86 disassembly, so I've written a patch to use Zydis in the case of x86, and Capstone in the case of ARM.

This doesn't fix the issue that Zydis simply shows the `AMX_TILE` category, though; I'll have to add that to Zydis at some point.

**Comment by grahamwoodward** [comment]
> Scratch that: Capstone folks seem to want to simply call Zydis to do their x86 disassembly, so I've written a patch to use Zydis in the case of x86, and Capstone in the case of ARM.
> 
> This doesn't fix the issue that Zydis simply shows the `AMX_TILE` category, though; I'll have to add that to Zydis at some point.

What?!?! Capstone wanna use Zydis?!?!

**Comment by matthew-olson-intel** [comment]
Well, it's just a proposal at the moment.

https://github.com/capstone-engine/capstone/discussions/2505

**Comment by matthew-olson-intel** [comment]
Now that we're back to using Zydis on the x86 side, to solve this issue permanently, we can categorize instructions by one of three things:

https://github.com/zyantific/zydis/blob/master/include/Zydis/MetaInfo.h

Category, ISA set, and ISA extension. Category is the most general, but aggregates all AMX instructions into `AMX_TILE`.

ISA set is more specific, but splits AVX512 into lots of little sub-categories:
```
    "AVX512BW_128",
    "AVX512BW_128N",
    "AVX512BW_256",
    "AVX512BW_512",
    "AVX512BW_KOP",
    "AVX512CD_128",
    "AVX512CD_256",
    "AVX512CD_512",
    "AVX512DQ_128",
    "AVX512DQ_128N",
    ...
```

I think what we might want is ISA extension, which seems to split the various extensions into smaller categories:
```
    "AMX_BF16",
    "AMX_FP16",
    "AMX_INT8",
    "AMX_TILE",
    "AVX",
    "AVX2",
    "AVX2GATHER",
    "AVX512EVEX",
    "AVX512VEX",
    ...
```

I think we can make this the default, and if users so choose, they can use categories instead (to see a good breakdown of "regular," non-extension, instructions).

**Comment by matthew-olson-intel** [comment]
@koleini I've implemented this in the `isa_ext` branch. You can pass `-e` to show ISA extensions, and if you'd like to look at "base" instructions compared to AMX INT8 and BF16, you can do:
```
-e -f BASE -f AMX_INT8 -f AMX_BF16
```
Let me know how it works for you!

**Comment by matthew-olson-intel** [comment]
@koleini Does this work, or do I need to do more figurin'?

**Comment by koleini** [comment]
@matthew-olson-intel works perfect, thanks!

**Comment by matthew-olson-intel** [comment]
@koleini Nice. Thanks for testing; I'll merge that then.
